celery/worker/job.py

```diff
@@ -437,8 +437,14 @@ def on_failure(self, exc_info):
                     self._announce_revoked(
                         'terminated', True, string(exc), False)
                     send_failed_event = False  # already sent revoked event
+
             # (acks_late) acknowledge after result stored.
-            if self.task.acks_late:
+            if self.task.acks_late and isinstance(exc, WorkerLostError):
+                #: For acks_late set to True, we changed the default behavior
+                #: to handle worker crash. Currently, we allow the message to be rejected
+                #: and requeued so it will be executed again by another worker.
```

Comment:
@grappeq do we want to change the behavior for all our tasks or do we prefer to add the property that will be a flag (like original reject_on_worker_lost) whether this should be used (what should be a default of the flag is another discussion)

Reply:
I would change this for all acks_late tasks as I cannot think of scenario where keeping the default behavior is preferable. Also, other tech leads are aware of the change we are making and they have not requested keeping this behavior.

Reply:
ok, I wanted to confirm that we're that brave :D